### PR TITLE
Fixing the page launcher

### PR DESCRIPTION
**Pull request by gregwoo-microsoft**
Still hitting bugs

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by gregwoo-microsoft** [comment]
Please note that this is still a work in progress.

File: WinUIGallery/App.xaml.cs [hunk 1]
```diff
@@ -167,87 +166,196 @@ private async void EnsureWindow(IActivatedEventArgs args = null)
...
+
             Type targetPageType = typeof(NewControlsPage);
             string targetPageArguments = string.Empty;
 
-            if (args != null)
+            if (args.Kind == ActivationKind.Launch)
             {
-                if (args.Kind == ActivationKind.Launch)
+                targetPageArguments = ((Windows.ApplicationModel.Activation.LaunchActivatedEventArgs)args).Arguments;
+            }
```
Comment (chingucoding) [inline WinUIGallery/App.xaml.cs:194]: It seems that in my testing, the Kind was always set to ActivationKind.Launch which is definitely not correct when using the protocol. I think there might be issues in the framework that prevent this from working properly.

File: WinUIGallery/App.xaml.cs [hunk 2]
```diff
@@ -167,87 +166,196 @@ private async void EnsureWindow(IActivatedEventArgs args = null)
...
             }
 
-            NavigationRootPage rootPage = StartupWindow.Content as NavigationRootPage;
-            rootPage.Navigate(targetPageType, targetPageArguments);
+            rootFrame.Navigate(targetPageType, targetPageArguments);
 
             if (targetPageType == typeof(NewControlsPage))
             {
-                ((Microsoft.UI.Xaml.Controls.NavigationViewItem)((NavigationRootPage)App.StartupWindow.Content).NavigationView.MenuItems[0]).IsSelected = true;
+                ((Microsoft.UI.Xaml.Controls.NavigationViewItem)((NavigationRootPage)Window.Current.Content).NavigationView.MenuItems[0]).IsSelected = true;
```
Comment (chingucoding) [inline WinUIGallery/App.xaml.cs:241]: Window.Current is null, I would continue using App.StartupWindow.

File: WinUIGallery/App.xaml.cs [hunk 3]
```diff
@@ -167,87 +166,196 @@ private async void EnsureWindow(IActivatedEventArgs args = null)
...
             }
             else if (targetPageType == typeof(ItemPage))
             {
-                NavigationRootPage.GetForElement(this).EnsureNavigationSelection(targetPageArguments);
+                NavigationRootPage.Current.EnsureNavigationSelection(targetPageArguments);
             }
 
             // Ensure the current window is active
-           StartupWindow.Activate();
+            Window.Current.Activate();
```
Comment (chingucoding) [inline WinUIGallery/App.xaml.cs:249]: Same here.

**Comment by chingucoding** [comment]
Superseded by #1304

**Pull request closed without merging**